app/src/main/java/fr/free/nrw/commons/contributions/ContributionsActivity.java

```diff
@@ -268,24 +270,21 @@ public int getTotalMediaCount() {
     }
 
     private void setUploadCount() {
-        UploadCountClient uploadCountClient = new UploadCountClient();
         CommonsApplication application = CommonsApplication.getInstance();
-        ListenableFuture<Integer> future = uploadCountClient
-                .getUploadCount(application.getCurrentAccount().name);
-        Futures.addCallback(future, new FutureCallback<Integer>() {
-            @Override
-            public void onSuccess(Integer uploadCount) {
-                getSupportActionBar().setSubtitle(getResources()
-                        .getQuantityString(R.plurals.contributions_subtitle,
-                                uploadCount,
-                                uploadCount));
-            }
-
-            @Override
-            public void onFailure(@NonNull Throwable t) {
-                Timber.e(t, "Fetching upload count failed");
-            }
-        }, ExecutorUtils.uiExecutor());
+
+        compositeDisposable.add(
+                RxJava2Tasks.getUploadCount(application.getCurrentAccount().name)
```

Comment:
I was also thinking about that. 
The thing is that MediaWiki is implemented by `ApacheHttpClientMediaWikiApi`, using autogenerated code which requires the deprecated ``ApacheHttp... I didn't want to mix this with that class :)

I'm unsure about how to better proceed with it. I would really want to deprecate `ApacheHttpClientMediaWikiApi` and stop using multiple http clients...
If we put this kind of new code into that class, wouldn't it contribute to making it last a bit longer?